apiserver/src/main/java/org/dependencytrack/resources/v1/AdvisoriesResource.java

```diff
@@ -0,0 +1,380 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.dependencytrack.resources.v1;
+
+import alpine.common.logging.Logger;
+import alpine.server.auth.PermissionRequired;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.headers.Header;
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.security.SecurityRequirements;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.glassfish.jersey.media.multipart.FormDataContentDisposition;
+import org.glassfish.jersey.media.multipart.FormDataParam;
+import io.csaf.retrieval.RetrievedDocument;
+import org.dependencytrack.datasource.vuln.csaf.CsafSource;
+import org.dependencytrack.datasource.vuln.csaf.ModelConverter;
+import org.dependencytrack.parser.dependencytrack.BovModelConverter;
+import org.dependencytrack.model.VulnerableSoftware;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.time.Instant;
+
+import org.dependencytrack.auth.Permissions;
+import org.dependencytrack.model.Advisory;
+import org.dependencytrack.model.Project;
+import org.dependencytrack.model.Vulnerability;
+import org.dependencytrack.model.validation.ValidUuid;
+import org.dependencytrack.persistence.QueryManager;
+import org.dependencytrack.persistence.jdbi.AdvisoryDao;
+import org.dependencytrack.resources.AbstractApiResource;
+import org.dependencytrack.resources.v1.openapi.PaginatedApi;
+import org.dependencytrack.resources.v1.problems.ProblemDetails;
+
+import java.util.List;
+
+import static org.dependencytrack.persistence.jdbi.JdbiFactory.withJdbiHandle;
+import static org.dependencytrack.resources.v1.FindingResource.MEDIA_TYPE_SARIF_JSON;
+
+/**
+ * JAX-RS resources for advisories.
+ *
+ * @author Lawrence Dean
+ * @since TODO set version
+ */
+@Path("/v1/advisories")
```

Comment:
v1 of the API uses the singular form of nouns. In [v2](https://github.com/DependencyTrack/hyades-apiserver/blob/main/api/src/main/openapi/openapi.yaml) we have adopted the (more common) plural form.

On that note, it would be ideal if new endpoints were added to v2, as it makes quality control easier with linting and breaking change detection etc.